src/Normalizer/ResponseOAuthClientCredentialsNormalizer.php

```diff
@@ -27,47 +27,53 @@ public function supportsNormalization($data, $format = null)
 
     public function denormalize($data, $class, $format = null, array $context = [])
     {
-        if (!is_object($data)) {
-            return null;
+        $object = new ResponseOAuthClientCredentials();
+
+        if (null === $data || false === \is_array($data)) {
+            return $object;
         }
 
-        $object = new ResponseOAuthClientCredentials();
+        if (\array_key_exists('token_type', $data) && $data['token_type'] !== null) {
+            $object->setTokenType($data['token_type']);
+            unset($data['token_type']);
+        }
 
-        if (property_exists($data, 'token_type') && $data->token_type !== null) {
-            $object->setTokenType($data->token_type);
+        if (\array_key_exists('expires_in', $data) && $data['expires_in'] !== null) {
+            $object->setExpiresIn($data['expires_in']);
+            unset($data['expires_in']);
         }
 
-        if (property_exists($data, 'expires_in') && $data->expires_in !== null) {
-            $object->setExpiresIn($data->expires_in);
+        if (\array_key_exists('access_token', $data) && $data['access_token'] !== null) {
+            $object->setAccessToken($data['access_token']);
+            unset($data['access_token']);
         }
 
-        if (property_exists($data, 'access_token') && $data->access_token !== null) {
-            $object->setAccessToken($data->access_token);
+        if (\array_key_exists('refresh_token', $data) && $data['refresh_token'] !== null) {
+            $object->setRefreshToken($data['refresh_token']);
+            unset($data['refresh_token']);
         }
 
-        if (property_exists($data, 'refresh_token') && $data->refresh_token !== null) {
-            $object->setRefreshToken($data->refresh_token);
+        foreach ($data as $key => $value) {
+            if (preg_match('/.*/', (string) $key)) {
+                $object[$key] = $value;
+            }
         }
 
         return $object;
     }
 
     public function normalize($object, $format = null, array $context = [])
     {
-        $data = new \stdClass();
-        if (null !== $object->getTokenType()) {
-            $data->token_type = $object->getTokenType();
+        $data = array();
+        $data['token_type'] = $object->getTokenType();
+        $data['expires_in'] = $object->getExpiresIn();
+        $data['access_token'] = $object->getAccessToken();
+        $data['refresh_token'] = $object->getRefreshToken();
+        foreach ($object as $key => $value) {
+            if (preg_match('/.*/', (string) $key)) {
+                $data[$key] = $value;
+            }
         }
```

Comment:
What is the purpose of these lines @avikarshasaha? My understanding here is that the regex will match any string, so it's checking to ensure all keys are strings? Seems pointless as we just finished defining the array keys above. Any thoughts on this @fedys, am I seeing this correctly?

Reply:
@dongilbert @fedys this conditions are meaningless for this normalizer, but I included them for symmetry. 
https://github.com/acquia/mc-cf-mega-openapi-php-sdk/blob/6c31a518de52070ff06cdc505ce0d5a47cf69e21/generated/Normalizer/AccountNormalizer.php#L111

Removing those lines!

Reply:
@avikarshasaha Taking a second look, those lines aren't for matching the $data array we just defined, but rather the keys of the object properties. Any object property key that is a string gets added to the $data array. I think this is in place to prevent private object properties (which would contain a null byte) from being assigned to the data array. 

I would retain that code for consistency with the generated code base.

Reply:
I tried this simple block of code:
```php
<?php

declare(strict_types=1);
error_reporting(-1);

class Some
{
    private $a   = 'a';
    protected $b = 'b';
    public $c    = 'c';
}

$some = new Some();
$data = [];

foreach ($some as $key => $value) {
    if (preg_match('/.*/', (string) $key)) {
        $data[$key] = $value;
    }
}

var_dump($data);
```
and the `$data` variable always contains just `['c' => 'c']` regardless of existence of the regex condition. But I'm okay with keeping the extra code.